src/proof/verify.rs

```diff
@@ -0,0 +1,217 @@
+//! Proof verification logic.
+
+use crate::{
+    nodes::{rlp_node, word_rlp, TrieNode, CHILD_INDEX_RANGE},
+    proof::ProofVerificationError,
+    EMPTY_ROOT_HASH,
+};
+use alloc::vec::Vec;
+use alloy_primitives::{Bytes, B256};
+use alloy_rlp::Decodable;
+use nybbles::Nibbles;
+
+/// Verify the proof for given key value pair against the provided state root.
+/// Returns the leaf node value for the given key.
+pub fn verify_proof<'a, I>(
+    proof: I,
+    root: B256,
+    key: B256,
```

Comment:
API should be: fn(root: B256, key: B256, value: Option<Vec<u8>>, proof: I) -> Ok(())